### DISCUSSION TITLE
Uvicorn Reload Getting Stuck : WARNING StarReload detected Changes in report_parser.py. Reloading....

### DISCUSSION THREAD

**Discussion by HarsimranjitSingh** [fastapi/fastapi#13054] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Uvicorn Reload Getting Stuck : WARNING StarReload detected Changes in file.py. Reloading....
```

### Description

I am running Uvicorn with --reload mode and it is randomly and very frequently getting stuck on below line.

Uvicorn Reload Getting Stuck : WARNING StarReload detected Changes in file.py. Reloading....

I need to kill using CRTL+C and start again at new PORT and it is randomly getting stuck when changes are made to any PY file

Any suggestion for overcome this issue :)

### Operating System

Windows

### FastAPI Version

0.115.6

### Pydantic Version

2.10.3

### Python Version

3.11.7

**Comment by luzzodev** [comment, marked as answer]
Hello @HarsimranjitSingh, 
if you are on windows and you are on pycharm this is a known bug from pycharm.
You can read more on it
https://github.com/encode/uvicorn/issues/2000
https://youtrack.jetbrains.com/issue/PY-60962/os.killpid-signal.CRTLCEVENT-ignored-when-running-python-program-from-Pycharm

try to start your app without pycharm.
